### PR TITLE
Compatilibity with Boost 1.65

### PR DESCRIPTION
**Pull request by jakobwilm**
Newer Boost versions such as 1.65 (default Ubuntu 18.04) declare the virtual clone method in error_info_base. This makes this class abstract and not compile.

### CODE REVIEW

File: src/cctag/utils/Exceptions.hpp [hunk 1]
```diff
@@ -73,6 +73,8 @@ class error_info<Tag, error_info_sstream>: public exception_detail::error_info_b
 		_value._v << value;
 	}
 
+    	error_info_base * clone() const {}
```
Comment (fabiencastan) [inline src/cctag/utils/Exceptions.hpp]: Are we not supposed to do something like that?
```
error_info_base * clone() const
{
  error_info* p = new error_info();
  *p = *this;
  return p;
}
```
To create a new instance with the same content.

Comment (jakobwilm) [reply src/cctag/utils/Exceptions.hpp]: Yes, absolutely. That looks correct.

**Comment by simogasp** [comment]
FYI in #78 I'm gonna get rid of all boost exception stuff and use simple std::exception.
We can merge this fix till then

**Comment by fabiencastan** [comment]
Created a new PR #82 with the discussed fix.

**Pull request closed without merging**